### PR TITLE
fix(grid.py) with rotation need to identify the upper left parent cel…

### PR DESCRIPTION
**Pull request by mnfienen**
…l as xy location with smallest x. this controls where the snapped xul.yul (relative to parent grid) ends up.

@aleaf can you take a look at this? Should make no change for rotation==0, but if rotated grid, can't rely on minx and miny of the upper left corner parent call to ID the xul/yul. Instead, need to get `argmin` for the farthest west (lowest X) vertex. 

Also, if the xul,yul calculated from xoff/yoff happens to land on a vertex, `modelgrid.intersect` can return the wrong parent cell pi,pj, so adding 0.01*delc to the xul coordinate to force pi,pj to be in the intended parent cell.

Hopefully no unintended consequences

### CODE REVIEW

File: mfsetup/grid.py [hunk 1]
```diff
@@ -728,9 +728,18 @@ def setup_structured_grid(xoff=None, yoff=None, xul=None, yul=None,
...
-        verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
-        xul, yul = verts[:, 0].min(), verts[:, 1].max()
+        # navigating the vertices of the parent cell depends on rotation angle
+        if rotation == 0:
+            pi, pj = parent_model.modelgrid.intersect(xul, yul)
+            verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
+            xul, yul = verts[:, 0].min(), verts[:, 1].max()
+        else:
+            # if there is rotation, need to make sure we locate within the actual upper left.
+            pi, pj = parent_model.modelgrid.intersect(xul + (delc_grid/100), yul)
```
Comment (aleaf) [inline mfsetup/grid.py:738]: what about using the cell center here (``xul + 0.5 * delr_grid, yul + 0.5 * delc_grid``)? That the corner will end up aligning with whatever parent cell the inset cell center is within. Also I think we want to use delr for x (spacing *along* a row) and delc for y (spacing *along* a column).

File: mfsetup/grid.py [hunk 2]
```diff
@@ -728,9 +728,18 @@ def setup_structured_grid(xoff=None, yoff=None, xul=None, yul=None,
...
+            verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
+            # upper left vertex has smallest x coordinate under rotation
+            
+            xul, yul = verts[np.argmin(verts[:,0])]
```
Comment (aleaf) [inline mfsetup/grid.py:742]: For clockwise rotation though, won't the smallest x be the *lower left* corner of the parent cell? Is there a way to get the cell vertices in the model coordinate system? That would allow the upper left of the cell to be identified, regardless if the rotation is CW or CCW.

**Review by aleaf — CHANGES_REQUESTED** [review]
@mnfienen thanks for catching this! A few thoughts:

File: mfsetup/grid.py (hunk 1 above)
Comment (mnfienen) [reply mfsetup/grid.py:738]: ugh - always  getting burned by the `delr` `delc` meaning. but....your suggestion can end up not in the center and going too high in y-space, resulting in snapping to the wrong cell. I think we need only a tiny nudge along only one axis (x-axis works for sure for CCW rotation).

File: mfsetup/grid.py (hunk 2 above)
Comment (mnfienen) [reply mfsetup/grid.py:742]: there is a `get_local_coords(x,y)` method on the flopy [grid](https://flopy.readthedocs.io/en/3.3.2/source/flopy.discretization.grid.html) object. I guess we could grab all the vertex coords, find the upper left (minx, maxy) and then flip back to world coords using `get_coords(x,y)`. I'll try that out. Perhaps we could also use that to choose `pi` and `pj` above.

**Comment by mnfienen** [comment]
OK - by flipping everything into model coords first, was able to generalize to work with rotated and unrotated grids (thanks for the idea @aleaf! I was struggling to think of how one would locate that corner regardless of rotation direction. I think we need a test but will have to do some shenanigans including making a negative-rotation angle version maybe of shellmound?

**Comment by mnfienen** [comment]
Shoot! This seems to have broken an SFR test. Doesn't seem actually related to the grid changes, though. Ideas @aleaf 

```        # this is overly specific,
        # and will break if the sfr package (inset extent, grid spacing, etc.) changes
>       assert to_parent == {23: 8, 25: 1}
E       assert {23: 10, 25: 1} == {23: 8, 25: 1}
E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {23: 10} != {23: 8}
E         Full diff:
E         - {23: 8, 25: 1}
E         ?      ^
E         + {23: 10, 25: 1}
E         ?      ^^

/home/runner/work/modflow-setup/modflow-setup/mfsetup/tests/test_lgr.py:152: AssertionError
